tests/test_esm_runscripts/test_helpers.py

```diff
@@ -0,0 +1,54 @@
+import unittest
+
+import esm_runscripts
+
+
+class TestHelpersUpdateReusableFiles(unittest.TestCase):
+    def setUp(self) -> None:
+        self.config = {
+            "general": {
+                "verbose": True,
+                "command_line_config": {
+                    "update_filetypes": [],
+                },
+                "potentially_reusable_filetypes": ["bin", "log", "src", "forcing"],
+                "reusable_filetypes": ["bin", "log", "src"],
+            },
+        }
+        return super().setUp()
+
+    def test_cli_bin_in_general(self) -> None:
+        """ "
+        Mimic a call such as::
+
+            $ esm_runscripts run.yaml -e exp1 --update-filetypes bin
+        """
+        # NOTE(PG): It would be nice to find a way to call the actual cli code here...
+        self.config["general"]["command_line_config"]["update_filetypes"] = ["bin"]
+        out_config = esm_runscripts.helpers.update_reusable_filetypes(self.config)
+        self.assertNotIn("bin", out_config["general"]["reusable_filetypes"])
+
+    def test_cli_extra_in_model(self) -> None:
+        """
+        Mimic a call such as::
+
+            $ esm_runscripts run.yaml -e exp1 --update-filetypes extra
+
+        Where the run.yaml has a model ``foo_model`` with the following
+        ``reusable_fletypes``::
+
+            foo_model:
+               reusable_filetypes: ["input", "forcing", "extra"]
+        """
+        self.config["general"]["command_line_config"]["update_filetypes"] = ["extra"]
+        self.config["general"]["potentially_reusable_filetypes"].append("extra")
+        self.config["foo_model"] = {}
+        self.config["foo_model"]["reusable_filetypes"] = ["input", "forcing", "extra"]
+        out_config = esm_runscripts.helpers.update_reusable_filetypes(
+            self.config, self.config["foo_model"]["reusable_filetypes"]
+        )
+        self.assertNotIn("extra", out_config)
```

Comment:
Shouldn't this be:
```suggestion
        self.assertNotIn("extra", out_config["genereal"]["reusable_filetypes"])
```

Reply:
I think that is correct, and that was the problem why the previous test was failing. When you give the `reusable_filetypes` as an input to the `update_reusable_filetypes` method it returns the list, not a dictionary.
https://github.com/esm-tools/esm_tools/blob/aa221c98f9a0d90511a57a1bf9abf0f04aadd76b/src/esm_runscripts/helpers.py#L174-L179
https://github.com/esm-tools/esm_tools/blob/aa221c98f9a0d90511a57a1bf9abf0f04aadd76b/src/esm_runscripts/helpers.py#L226-L228
This is because this method is used in different places in the code:
1. To change the general one (no `reusable_filetypes` provided)
2. In `filelists.py` where it is used for each model and for that case, the files that are to be reused are calculated on the fly and don't make it into the config dictionary.

Reply:
OK, I renamed the output variable of the function to hopefully make more sense. Once the tests go through, I'll merge this.